modules/internal/ChapelSyncvar.chpl

```diff
@@ -803,9 +831,21 @@ module ChapelSyncvar {
     return new _singlevar(from);
   }
 
+  proc chpl__compilerGeneratedAssignSyncSingle(ref lhs : _singlevar(?),
+                                               ref rhs : _singlevar(?)) {
+    // TODO: Should this clone the value and the full/empty state instead?
+    lhs.writeEF(rhs.readFF());
+  }
+
+  proc chpl__compilerGeneratedCopySyncSingle(ref sv : _singlevar(?)) {
+    // TODO: this should probably clone the value and full/empty state instead
```

Comment:
@mppf With respect to this TODO and the one just above:  For singles, these effectively are just cloning the full/empty state, right?  Or do you mean something more technical by "clone"?

Not critical for this PR, but I'm also curious what your thoughts are about the post-deprecation status of these routines.  This is arguably where I got hung up on my effort...  it didn't seem right to use a sync/single to initialize a sync/single which is what led me to try to pipe `t` values through initializers for fields of type `sync t`, so I got bogged down in trying to make that work rather than preserving the status quo.

It looks to me like perhaps only 1 test other than your new deprecation tests actually calls into these routines at execution time, and only ~18 others have the compiler generate them, but then don't use them.

Reply:
@bradcray - 

>  For singles, these effectively are just cloning the full/empty state, right? Or do you mean something more technical by "clone"?

I might be misunderstanding your comment, but

``` chapel
    var ret: sv.type = sv.readFF();
```

doesn't clone the full/empty state - rather, it insists that the full/empty state is full and then "clones" the value.
If we were cloning the full/empty state, I would expect that - if the input variable `sv` is empty - then the output variable will also be empty.

I'm not really sure where the future will take us here. It seems like we can copy-initialize sync/single variables but it's not really clear which meaning we should give it (clone-allowing-empty vs the current wait-until-full-then-clone). I noted this as Future Work in the PR description though - and I think it's worth noting that the issue is present both for compiler-generated initializers as well as arrays or other records containing sync/single. In both cases we want to run some sort of default copy operation but it's not so clear what that is for sync/single.

Anyway, I suspect that clone-allowing-empty is what we want and won't be surprising, but I am happy we are not trying to do it in this PR.

Reply:
> If we were cloning the full/empty state, I would expect that - if the input variable sv is empty - then the output variable will also be empty.

Ah, I see.  That makes sense, thanks.